liquibase-core/src/main/java/liquibase/changelog/DatabaseChangeLog.java

```diff
@@ -656,6 +659,13 @@ public void includeAll(String pathName,
                 if (StringUtil.isNotEmpty(path) && !(path.endsWith("/"))) {
                     path = path + '/';
                 }
+
+                if (ChangeLogParserConfiguration.ERROR_ON_CIRCULAR_INCLUDE_ALL.getCurrentValue()) {
+                    if (seenChangelogPaths.contains(path)) {
+                        throw new SetupException("Circular reference detected in '" + path + "'. Set " + ChangeLogParserConfiguration.ERROR_ON_CIRCULAR_INCLUDE_ALL.getKey() + " if you'd like to ignore this error.");
```

Comment:
So.. if we are adding a feature to disable this validation, should we call this an error? Maybe s/error/setting/g ?